types/k6/experimental/browser.d.ts

```diff
@@ -0,0 +1,1508 @@
+      x: number;
+
+      y: number;
+    };
+
+    /**
+     * When `true`, the call requires selector to resolve to a single element.
+     * If given selector resolves to more than one element, the call throws
+     * an exception. Defaults to `false`.
+     */
+    strict?: boolean;
+
+    /**
+     * Maximum time in milliseconds. Defaults to `0` - no timeout. Default is
+     * overridden by the `setDefaultTimeout` option on `BrowserContext` or
+     * `page` methods.
+     */
+    timeout?: number;
+
+    /**
+     * Setting this to `true` will perform the actionability checks without
+     * performing the action. Useful to wait until the element is ready for the
+     * action without performing it. Defaults to `false`.
+     */
+    trial?: boolean;
+  }): void;
+
+  /**
+   * Returns the page's URL.
+   */
+  url(): string;
+
+  /**
+   * Returns the page's size (width and height).
+   */
+  viewportSize(): {
+    /**
+     * page width in pixels.
+     */
+    width: number;
+
+    /**
+     * page height in pixels.
+     */
+    height: number;
+  };
+
+  /**
+   * Returns when the `pageFunction` returns a truthy value.
+   *
+   * @param pageFunction Function to be evaluated in the page context.
+   * @param arg Optional argument to pass to `pageFunction`.
+   * @param options
+   */
+  waitForFunction<R, Arg>(pageFunction: PageFunction<Arg, R>, options?: {
+    /**
+     * If `polling` is `'raf'`, then `pageFunction` is constantly executed in
+     * `requestAnimationFrame` callback. If `polling` is a number, then it is
+     * treated as an interval in milliseconds at which the function would be
+     * executed. Defaults to `raf`.
+     */
+    polling?: number|"raf";
+
+    /**
+     * Maximum time in milliseconds. Defaults to `0` - no timeout. Default is
+     * overridden by the `setDefaultTimeout` option on `BrowserContext` or
+     * `page` methods.
+     */
+    timeout?: number;
+  }, arg?: Arg): Promise<JSHandle<R>>;
+
+  /**
+   * This waits for the given load state to be reached. It will immediately
+   * unblock if that lifecycle event has already been received.
+   *
+   * @param state Optional load state to wait for, defaults to `load`:
+   * - `'domcontentloaded'` - consider operation to be finished when the
+   * `DOMContentLoaded` event is fired.
+   * - `'load'` - consider operation to be finished when the `load` event is
+   * fired.
+   * - `'networkidle'` - **DISCOURAGED** consider operation to be finished
+   * when there are no network connections for at least `500` ms. Don't use
+   * this method for testing especially with chatty websites where the event
+   * may never fire, rely on web assertions to assess readiness instead.
+   * @param options
+   */
+  waitForLoadState(state?: "load"|"domcontentloaded"|"networkidle", options?: {
+    /**
+     * Maximum operation time in milliseconds. Defaults to `0` - no timeout. The
+     * default value can be changed via `navigationTimeout` option in the config,
```

Comment:
I believe this `navigationTimeout` option in the config does not apply to our implementation right?